src/components/Select/Select.jsx

```diff
@@ -526,6 +526,7 @@ export default class Select extends React.Component {
               keyMap={keyMap}
               hasSelectedItems={some(items, { selected: true })}
               onClick={event => this.handleSelectOption(item, event)}
+              onItemClick={() => this.handleSelectOption(item, item.onItemClick)}
```

Comment:
I just moved the item.onItemClick here, as it was doing the action but it wasn't closing it because it never got to this point.